### PR TITLE
Update simapp

### LINKED ISSUE #363: Use protobuf encoded public key in BaseAccount
Once https://github.com/cosmos/cosmos-sdk/issues/6886 is available in simapp.

This code needs to be adapted:

```ts
function decodeBaseAccount(data: Uint8Array, prefix: string): Account {
  const { address, pubKey, accountNumber, sequence } = cosmos.auth.BaseAccount.decode(data);
  // Pubkey is still Amino-encoded in BaseAccount (https://github.com/cosmos/cosmos-sdk/issues/6886)
  const pubkey = pubKey.length ? decodeAminoPubkey(pubKey) : null;
  return {
    address: Bech32.encode(prefix, address),
    pubkey: pubkey,
    accountNumber: uint64FromProto(accountNumber).toNumber(),
    sequence: uint64FromProto(sequence).toNumber(),
  };
}
```

### LINKED ISSUE #377: Update simapp
Let's follow Cosmos SDK development step by step, adapting our code as needed along the way.

- [x] to `nightly-2020-08-11` and `REF=20c80cfd44a068a592148d648c50de3459fd6295` #381 
- [ ] to `nightly-2020-08-12` and `REF=7ea64676788f2d550a65d86cd6b4cdfe89e9a49b` (blocked by https://github.com/cosmos/cosmos-sdk/issues/7026)
- [ ] to `nightly-2020-08-13` (blocked by https://github.com/cosmos/cosmos-sdk/issues/7026)
- [ ] to `nightly-2020-08-14` (blocked by https://github.com/cosmos/cosmos-sdk/issues/7026)
- [ ] to `nightly-2020-08-15` (blocked by https://github.com/cosmos/cosmos-sdk/issues/7026) (requires Tendermint 0.34)
- [ ] to `nightly-2020-08-16` (requires Tendermint 0.34)
- [ ] to `nightly-2020-08-17` (requires Tendermint 0.34)
- [ ] to `nightly-2020-08-18` (requires Tendermint 0.34)
- [ ] to `nightly-2020-08-19` (requires Tendermint 0.34)

### LINKED ISSUE #383: Submit TxRaw instead of Tx
IIRC the spec says transactions are boradcasted as TxRaw, but backend seems to to reject those. Need to check the spec again and the Cosmos SDK state.

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #363 
Closes #377
Closes #383 

I wasn’t sure where to put the public key encoding/decoding. Launchpad?

### CODE REVIEW

**Review by webmaster128 — COMMENTED** [review]
Beautiful.

Do the tests pass when broadcasting a TxRaw? Then this also closes https://github.com/CosmWasm/cosmjs/issues/383

> where to put the public key encoding/decoding. Launchpad?

I'd put it in proto-signing. Maybe not perfect, but that is shares between the packages where we need it. Launchpad should not need to care about protobuf at all.

**Comment by willclarktech** [comment]
@webmaster128 Does this PR need to support pubkey types besides secp256k1?

**Comment by webmaster128** [comment]
> @webmaster128 Does this PR need to support pubkey types besides secp256k1?

Nope. As long as we have the framework that makes it clear where to add support, this is fine

File: packages/stargate/src/stargateclient.ts [hunk 1]
```diff
@@ -85,20 +85,19 @@ function uint64FromProto(input: number | Long | null | undefined): Uint64 {
   return Uint64.fromString(input.toString());
 }
 
-function accountFromProto(input: cosmos.auth.IBaseAccount, prefix: string): Account {
+function accountFromProto(input: cosmos.auth.v1beta1.IBaseAccount): Account {
   const { address, pubKey, accountNumber, sequence } = input;
-  // Pubkey is still Amino-encoded in BaseAccount (https://github.com/cosmos/cosmos-sdk/issues/6886)
-  const pubkey = pubKey && pubKey.length ? decodeAminoPubkey(pubKey) : null;
+  const pubkey = decodePubkey(pubKey);
```
Comment (webmaster128) [inline packages/stargate/src/stargateclient.ts:90]: This diff closes https://github.com/CosmWasm/cosmjs/issues/363, right?

Comment (willclarktech) [reply packages/stargate/src/stargateclient.ts:90]: Yes :)

File: packages/proto-signing/src/pubkey.spec.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import { google } from "./codec";
+import { decodePubkey, encodePubkey } from "./pubkey";
+
+const { Any } = google.protobuf;
+
+describe("pubkey", () => {
+  const defaultPubkeyBase64 = "AtQaCqFnshaZQp6rIkvAPyzThvCvXSDO+9AzbxVErqJP";
+  const defaultPubkeyBytes = fromBase64(defaultPubkeyBase64);
+  const defaultPubkeyProtoBytes = Uint8Array.from([0x0a, defaultPubkeyBytes.length, ...defaultPubkeyBytes]);
```
Comment (webmaster128) [inline packages/proto-signing/src/pubkey.spec.ts:12]: Any clue what kind of encoding this is, especially the leading `0x0a`?

**Review by webmaster128 — APPROVED** [review]

File: packages/proto-signing/src/pubkey.spec.ts (hunk 2 above)
Comment (willclarktech) [reply packages/proto-signing/src/pubkey.spec.ts:12]: I believe this is protobuf for "field 1; length-prefixed".

See https://github.com/protobufjs/protobuf.js/wiki/How-to-reverse-engineer-a-buffer-by-hand#so-what-are-those-wire-types

Comment (webmaster128) [reply packages/proto-signing/src/pubkey.spec.ts:12]: Ah, then this is https://github.com/cosmos/cosmos-sdk/blob/v0.40.0-rc0/proto/cosmos/crypto/secp256k1/keys.proto#L13

**Pull request merged**